File: msgraph-mail/python/models/size_range.py
from __future__ import annotations
from kiota_abstractions.serialization import AdditionalDataHolder, Parsable, ParseNode, SerializationWriter
from typing import Any, Callable, Dict, List, Optional, Union

class SizeRange(AdditionalDataHolder, Parsable):
    @property
    def additional_data(self,) -> Dict[str, Any]:
        """
        Gets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
        Returns: Dict[str, Any]
        """
        return self._additional_data

    @additional_data.setter
    def additional_data(self,value: Dict[str, Any]) -> None:
        """
        Sets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
        Args:
            value: Value to set for the AdditionalData property.
        """
        self._additional_data = value

    def __init__(self,) -> None:
        """
        Instantiates a new sizeRange and sets the default values.
        """
        # The maximum size (in kilobytes) that an incoming message must have in order for a condition or exception to apply.
        self._maximum_size: Optional[int] = None

        # The minimum size (in kilobytes) that an incoming message must have in order for a condition or exception to apply.
        self._minimum_size: Optional[int] = None

        # Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
        self._additional_data: Dict[str, Any] = {}


    @staticmethod
    def create_from_discriminator_value(parse_node: Optional[ParseNode] = None) -> SizeRange:
        """
        Creates a new instance of the appropriate class based on discriminator value
        Args:
            parseNode: The parse node to use to read the discriminator value and create the object
        Returns: SizeRange
        """
        if not parse_node:
            raise Exception("parse_node cannot be undefined")
        return SizeRange()

    def get_field_deserializers(self,) -> Dict[str, Callable[[ParseNode], None]]:
        """
        The deserialization information for the current model
        Returns: Dict[str, Callable[[ParseNode], None]]
        """
        fields = {
            "maximum_size": lambda n : setattr(self, 'maximum_size', n.get_int_value()),
            "minimum_size": lambda n : setattr(self, 'minimum_size', n.get_int_value()),
        }
        return fields

    @property
    def maximum_size(self,) -> Optional[int]:
        """
        Gets the maximumSize property value. The maximum size (in kilobytes) that an incoming message must have in order for a condition or exception to apply.
        Returns: Optional[int]
        """
        return self._maximum_size

    @maximum_size.setter
    def maximum_size(self,value: Optional[int] = None) -> None:
        """
        Sets the maximumSize property value. The maximum size (in kilobytes) that an incoming message must have in order for a condition or exception to apply.
        Args:
            value: Value to set for the maximumSize property.
        """
        self._maximum_size = value

    @property
    def minimum_size(self,) -> Optional[int]:
        """
        Gets the minimumSize property value. The minimum size (in kilobytes) that an incoming message must have in order for a condition or exception to apply.
        Returns: Optional[int]
        """
        return self._minimum_size

    @minimum_size.setter
    def minimum_size(self,value: Optional[int] = None) -> None:
        """
        Sets the minimumSize property value. The minimum size (in kilobytes) that an incoming message must have in order for a condition or exception to apply.
        Args:
            value: Value to set for the minimumSize property.
        """
        self._minimum_size = value

    def serialize(self,writer: SerializationWriter) -> None:
        """
        Serializes information the current object
        Args:
            writer: Serialization writer to use to serialize this model
        """
        if not writer:
            raise Exception("writer cannot be undefined")
        writer.write_int_value("maximumSize", self.maximum_size)
        writer.write_int_value("minimumSize", self.minimum_size)
        writer.write_additional_data_value(self.additional_data)

File: msgraph-mail/php/src/Models/SizeRange.php
<?php

namespace Microsoft\Graph\Models;

use Microsoft\Kiota\Abstractions\Serialization\AdditionalDataHolder;
use Microsoft\Kiota\Abstractions\Serialization\Parsable;
use Microsoft\Kiota\Abstractions\Serialization\ParseNode;
use Microsoft\Kiota\Abstractions\Serialization\SerializationWriter;

class SizeRange implements AdditionalDataHolder, Parsable 
{
    /**
     * @var array<string, mixed>|null $additionalData Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
    */
    private ?array $additionalData = null;
    
    /**
     * @var int|null $maximumSize The maximum size (in kilobytes) that an incoming message must have in order for a condition or exception to apply.
    */
    private ?int $maximumSize = null;
    
    /**
     * @var int|null $minimumSize The minimum size (in kilobytes) that an incoming message must have in order for a condition or exception to apply.
    */
    private ?int $minimumSize = null;
    
    /**
     * Instantiates a new SizeRange and sets the default values.
    */
    public function __construct() {
        $this->setAdditionalData([]);
    }

    /**
     * Creates a new instance of the appropriate class based on discriminator value
     * @param ParseNode $parseNode The parse node to use to read the discriminator value and create the object
     * @return SizeRange
    */
    public static function createFromDiscriminatorValue(ParseNode $parseNode): SizeRange {
        return new SizeRange();
    }

    /**
     * Gets the AdditionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @return array<string, mixed>|null
    */
    public function getAdditionalData(): ?array {
        return $this->additionalData;
    }

    /**
     * The deserialization information for the current model
     * @return array<string, callable(ParseNode): void>
    */
    public function getFieldDeserializers(): array {
        $o = $this;
        return  [
            'maximumSize' => fn(ParseNode $n) => $o->setMaximumSize($n->getIntegerValue()),
            'minimumSize' => fn(ParseNode $n) => $o->setMinimumSize($n->getIntegerValue()),
        ];
    }

    /**
     * Gets the maximumSize property value. The maximum size (in kilobytes) that an incoming message must have in order for a condition or exception to apply.
     * @return int|null
    */
    public function getMaximumSize(): ?int {
        return $this->maximumSize;
    }

    /**
     * Gets the minimumSize property value. The minimum size (in kilobytes) that an incoming message must have in order for a condition or exception to apply.
     * @return int|null
    */
    public function getMinimumSize(): ?int {
        return $this->minimumSize;
    }

    /**
     * Serializes information the current object
     * @param SerializationWriter $writer Serialization writer to use to serialize this model
    */
    public function serialize(SerializationWriter $writer): void {
        $writer->writeIntegerValue('maximumSize', $this->getMaximumSize());
        $writer->writeIntegerValue('minimumSize', $this->getMinimumSize());
        $writer->writeAdditionalData($this->getAdditionalData());
    }

    /**
     * Sets the AdditionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @param array<string,mixed> $value Value to set for the AdditionalData property.
    */
    public function setAdditionalData(?array $value): void {
        $this->additionalData = $value;
    }

    /**
     * Sets the maximumSize property value. The maximum size (in kilobytes) that an incoming message must have in order for a condition or exception to apply.
     * @param int|null $value Value to set for the maximumSize property.
    */
    public function setMaximumSize(?int $value): void {
        $this->maximumSize = $value;
    }

    /**
     * Sets the minimumSize property value. The minimum size (in kilobytes) that an incoming message must have in order for a condition or exception to apply.
     * @param int|null $value Value to set for the minimumSize property.
    */
    public function setMinimumSize(?int $value): void {
        $this->minimumSize = $value;
    }

}
